mysteriousorganism.js

```diff
@@ -0,0 +1,136 @@
+// Returns a random DNA base
+const returnRandBase = () => {
+	const dnaBases = ['A', 'T', 'C', 'G'];
+	return dnaBases[Math.floor(Math.random() * 4)];
+};
+
+// Returns a random single stand of DNA containing 15 bases
+const mockUpStrand = () => {
+	const newStrand = [];
+	for (let i = 0; i < 15; i++) {
+		newStrand.push(returnRandBase());
+	}
+	return newStrand;
+};
+
+// Create function factory
+const pAequorFactory = (specimenNum, dna) => {
+	return {
+		specimenNum,
+		dna,
+
+		mutate() {
+			const dnaBases = ['A', 'T', 'C', 'G']; // You'd probably want this variable to be global at some point.
```

Comment:
Wat me opvalt is dat je binnen de mutate method weer je eigen logic maakt om een random base te selecteren. Dat is zonde want regel 1 van dit document begint letterlijk met een functie die dat doet...

Reply:
Voorbeeld:
```
mutate() {
    mutateBaseIndex = Math.floor(Math.random() * (dna.length - 1));
    let newStrand = returnRandBase();
    while (newStrand == dna[mutateBaseIndex]) newStrand = returnRandBase();
    dna[mutateBaseIndex] = newStrand;
    return dna;
  }

Reply:
> General: In de for loops zou je de loop variabele beter met let kunnen beginnen. Dan heb je het hele scope gelul denk ik ook niet. dus:
> 
> `for(let i = 0; i < array.length; i++){}`
> 
> Zie: https://www.w3schools.com/js/js_loop_for.asp

Ja zo heb ik het ook geleerd inderdaad, is er doorheen geslipt maar dat verklaard de scope issues.

Reply:
> Wat me opvalt is dat je binnen de mutate method weer je eigen logic maakt om een random base te selecteren. Dat is zonde want regel 1 van dit document begint letterlijk met een functie die dat doet...

Maar de opdracht is om 1 van de tekens te veranderen, niet een hele nieuwe string te maken.